backend/danswer/connectors/alation/connector.py

```diff
@@ -0,0 +1,228 @@
+        make API requests.
+        """
+        response = self.api_client.post(
+            self.create_url("/integration/v1/createAPIAccessToken/"),
+            json={"user_id": self.user_id, "refresh_token": self.refresh_token},
+        )
+        data = response.json()
+        self.api_client.headers.update({"Token": data["api_access_token"]})
+        self.token_expiry = datetime.fromisoformat(data["token_expires_at"])
+
+    def _validate_access_token(self) -> None:
+        """Refreshes the access token if it expires while we're making requests
+        TODO - Probably needs to also check the validate API to handle expiring
+        refresh tokens.
+        """
+        if (
+            self.token_expiry is None
+            or datetime.now(self.token_expiry.tzinfo) >= self.token_expiry
+        ):
+            self.authenticate()
+
+    def get_batch_for_object_type(
+        self, otype: AlationObjectType, limit: int = 100, skip: int = 0
+    ) -> list[dict[str, Any]]:
+        """A simplified function for making batch requests to get Alation objects.
+        Not all Alation APIs work this way, but this'll do for current objects.
+        """
+        self._validate_access_token()
+        response = self.api_client.get(
+            self._create_url_for_object(
+                otype, params={"limit": str(limit), "skip": str(skip)}
+            ),
+        )
+        response.raise_for_status()
+        return response.json()
+
+    @staticmethod
+    def get_title_from_page(page: dict, otype: AlationObjectType) -> str:
+        """A helper to get the title for an Alation object"""
+        if otype == AlationObjectType.ARTICLE:
+            if len(page["title"]) == 0:
+                return f"Untitled Article [{page['id']}]"
+            return page["title"]
+        else:
+            if len(page["title"]) == 0:
+                return page["key"]
+            return page["title"]
+
+    @staticmethod
+    def get_body_from_page(page: dict, otype: AlationObjectType) -> str:
+        """A helper to get some kind of document body from an Alation object"""
+        if otype == AlationObjectType.ARTICLE:
+            return page["body"]
+        else:
+            return page["description"]
+
+
+class AlationConnector(LoadConnector):
+    """Connector to the Alation Data Catalog
+
+    Arguments:
+    server_url -- URL to the Alation Server
+    object_types_to_index -- List of Alation Object Types to index
+    batch_size -- The maximum number of objects to index in 1 request
+    max_objects_to_index_per_type -- The maximum number of objects to index for a given
+    object type. This is to reduce the liklihood of rate-limiting
+    """
+
+    def __init__(
+        self,
+        server_url: str,
+        object_types_to_index: list[AlationObjectType] = ALL_ALATION_OBJECT_TYPES,
+        batch_size: int = INDEX_BATCH_SIZE,
+        max_objects_to_index_per_type: int = 0,
```

Comment:
prefer to use None for uncapped. Also I may just be having a brain freeze moment here but I don't see why 0 is uncapped in the logic below?